lib/normalize.js

```diff
@@ -251,7 +251,7 @@ const normalize = async (pkg, { strict, steps, root, changes, allowLegacyCase })
 
   // strip "node_modules/.bin" from scripts entries
   // remove invalid scripts entries (non-strings)
-  if (steps.includes('scripts') || steps.includes('scriptpath')) {
+  if ((steps.includes('scripts') || steps.includes('scriptpath')) && data.scripts !== undefined) {
```

Comment:
Should we keep the `scripts` section if it's empty? 
Maybe align the logic with the `bin`: https://github.com/npm/package-json/blob/main/lib/normalize.js#L76 ?

Reply:
Yes we should.  We don't want to change behavior here unless we have to.  This behavior is very old and at this point we should assume that consumers of npm manifests are treating this attribute as always being there even if empty.